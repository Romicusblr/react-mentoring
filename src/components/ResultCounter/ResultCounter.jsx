import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './ResultCounter.module.css';

function ResultCounter({ quantity, className, ...other }) {
  return (
    <b
      className={classNames(style.resultCounter, className)}
      {...other}
    >
      {`${quantity} movie found`}
    </b>
  );
}

ResultCounter.propTypes = {
  quantity: PropTypes.number,
  className: PropTypes.string,
};

ResultCounter.defaultProps = {
  className: '',
  quantity: 0,
};

export default ResultCounter;
