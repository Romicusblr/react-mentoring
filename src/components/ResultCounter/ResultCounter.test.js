import ResultCounter from '.';

fdescribe('ResultCounter', () => {
  it('should be selectable by class "resultCounter"', () => {
    expect(shallow(<ResultCounter />).is('.resultCounter')).toBe(true);
  });

  it('should match to snapshot', () => {
    const snap = shallow(<ResultCounter />);
    expect(snap).toMatchSnapshot();
  });
});
