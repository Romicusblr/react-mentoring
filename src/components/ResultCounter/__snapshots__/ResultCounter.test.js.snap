// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCounter should match to snapshot 1`] = `
<b
  className="resultCounter"
>
  undefined movie found
</b>
`;
